### DISCUSSION TITLE
How can I trigger DJL to download engine's C++ library?

### DISCUSSION THREAD

**Discussion by SuperMaskv** [deepjavalibrary/djl#2183] [Q&A] [answered]
I'm trying to prepare everything while building docker image, how can I prepare native library for my app?

**Comment by frankfliu** [comment, marked as answer]
You can bundle the native library when you build your java project. We provide maven package for each platform to make it easy to distribute your application (or build docker image), for example, here is the document about how to install PyTorch engine for Linux CPU: https://docs.djl.ai/master/engines/pytorch/pytorch-engine/index.html#linux-cpu

You can also take a look our demo project how to package your application: https://github.com/deepjavalibrary/djl-demo/tree/master/development/fatjar

**Reply by SuperMaskv** [reply]
Thx a lot, it works.
